File: TheBitGame.py
#User function Template for python3

class Solution:
    def swapBitGame (self,N):
        # code here
        if len(bin(N).replace('0b', '').replace('0', '')) % 2 == 0: return 2
        return 1

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        
        N=int(input())
        

        ob = Solution()
        print(ob.swapBitGame(N))
# } Driver Code Ends

File: bit_manipulation/The_Bit_Game/cpp/the_bit_game.cpp
// { Driver Code Starts
#include <bits.h/stdc++.h>
#include<iostream>
using namespace std;
// } Driver Code Ends

class Solution
{
public:
    int swapBitGame(long long N)
    {
        // Code here
        int count = 0;
        while (N > 0)
        {
            if (N % 2 == 1)
            {
                count++;
            }
            N /= 2;
        }
        return (count % 2 == 0) ? 2 : 1;
    }
};

// { Driver Code Starts
int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        long long N;
        cin >> N;
        Solution ob;
        cout << ob.swapBitGame(N) << endl;
    }
    return 0;
}
// } Driver Code Ends